tequilapi/http_api_server.go

```diff
@@ -36,18 +67,35 @@ type APIServer interface {
 
 type apiServer struct {
 	errorChannel chan error
-	handler      http.Handler
 	listener     net.Listener
+
+	gin *gin.Engine
 }
 
 // NewServer creates http api server for given address port and http handler
-func NewServer(listener net.Listener, handler http.Handler, corsPolicy CorsPolicy) APIServer {
+func NewServer(
+	listener net.Listener,
+	handlers []func(e *gin.Engine) error,
+) (APIServer, error) {
+	g := gin.New()
+	g.Use(gin.Recovery())
+	g.Use(cors.New(corsConfig))
+
+	for _, h := range handlers {
+		err := h(g)
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	server := apiServer{
 		errorChannel: make(chan error, 1),
-		handler:      DisableCaching(ApplyCors(handler, corsPolicy)),
```

Comment:
Both `DisableCaching` and `ApplyCors` are mostly useless now. You think it's worth keeping it?